### PR TITLE
Add key filing dates to company report

### PR DESCRIPTION
**Pull request by leahtaylor**
Map key filing dates for company report generation

PCI-58

### CODE REVIEW

File: document-generator-company-report/src/main/java/uk/gov/companieshouse/document/generator/company/report/mapping/mappers/CompanyReportMapperDecorator.java [hunk 1]
```diff
@@ -1,15 +1,18 @@
 package uk.gov.companieshouse.document.generator.company.report.mapping.mappers;
 
 import java.util.List;
+import javax.swing.KeyStroke;
```
Comment (caismith-ch) [inline document-generator-company-report/src/main/java/uk/gov/companieshouse/document/generator/company/report/mapping/mappers/CompanyReportMapperDecorator.java]: Unused import

File: document-generator-company-report/src/main/java/uk/gov/companieshouse/document/generator/company/report/mapping/mappers/CompanyReportMapperDecorator.java [hunk 2]
```diff
@@ -38,6 +44,12 @@ public CompanyReport mapCompanyReport(CompanyReportApiData companyReportApiData)
             if (companyReportApiData.getCompanyProfileApi().getPreviousCompanyNames() != null) {
                 companyReport.setPreviousNames(setPreviousNames(companyReportApiData.getCompanyProfileApi().getPreviousCompanyNames()));
             }
+
+            if(companyReportApiData.getCompanyProfileApi().getAccounts() != null) {
+                if (companyReportApiData.getCompanyProfileApi().getAccounts() != null) {
```
Comment (caismith-ch) [inline document-generator-company-report/src/main/java/uk/gov/companieshouse/document/generator/company/report/mapping/mappers/CompanyReportMapperDecorator.java]: Looks as though this line could be removed as it's doing the same as the above line.

File: document-generator-company-report/src/main/java/uk/gov/companieshouse/document/generator/company/report/mapping/model/document/items/keyfilingdates/KeyFilingDates.java [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+
+        @JsonProperty("last_confirmation_statement")
+        private String lastConfirmationStatement;
+
+        @JsonProperty("next_confirmation_statement")
+        private String nextConfirmationStatement;
+
+        @JsonProperty("last_members_list")
+        private String lastMembersList;
+
```
Comment (caismith-ch) [inline document-generator-company-report/src/main/java/uk/gov/companieshouse/document/generator/company/report/mapping/model/document/items/keyfilingdates/KeyFilingDates.java]: The above variables seem to have been formatted to the right

File: document-generator-company-report/src/main/java/uk/gov/companieshouse/document/generator/company/report/mapping/mappers/keyfilingdates/ApiToKeyFilingDatesMapper.java [hunk 4]
```diff
@@ -1,6 +1,9 @@
 package uk.gov.companieshouse.document.generator.company.report.mapping.mappers.keyfilingdates;
 
+import java.text.ParseException;
```
Comment (jterrych) [inline document-generator-company-report/src/main/java/uk/gov/companieshouse/document/generator/company/report/mapping/mappers/keyfilingdates/ApiToKeyFilingDatesMapper.java]: Are these unused imports?

**Pull request merged**